components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/util/ResponseTypeHandlerUtil.java

```diff
@@ -253,9 +254,15 @@ public static AuthzCodeDO generateAuthorizationCode(OAuthAuthzReqMessageContext
         // set code issued time.this is needed by downstream handlers.
         oauthAuthzMsgCtx.setCodeIssuedTime(timestamp.getTime());
 
-        if (authorizationReqDTO.getUser() != null && authorizationReqDTO.getUser().isFederatedUser()) {
-            //if a federated user, treat the tenant domain as similar to application domain.
-            authorizationReqDTO.getUser().setTenantDomain(authorizationReqDTO.getTenantDomain());
+        AuthenticatedUser authenticatedUser = authorizationReqDTO.getUser();
+        if (authenticatedUser != null && authenticatedUser.isFederatedUser()) {
+            String appName = OAuth2Util.getServiceProvider(authorizationReqDTO.getConsumerKey()).getApplicationName();
```

Comment:
`OAuth2Util.getServiceProvider(authorizationReqDTO.getConsumerKey())` will this return null at any point?

Reply:
No, if the client ID is not valid then it will throw "InvalidOAuthClientException" & it won't become null
https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java#L3286